yjit_asm.h

```diff
@@ -55,8 +55,14 @@ typedef struct CodeBlock
     // Flag to enable or disable comments
     bool has_asm;
 
+    // Keep track of the current aligned write position.
+    // Used for changing protection when writing to the JIT buffer
+    uint32_t current_aligned_write_pos;
```

Comment:
What's the aligned write position? What does it mean???

Reply:
This variable is similar to the `write_pos` variable but it's aligned on multiples of the OS page size.  That way [when we write to the buffer, we can compare and only call the `mprotect` system call when we're on a "new" page](https://github.com/ruby/ruby/pull/5032/files#diff-b3441aa2144861250dbf49622387ff434c7a421c22f313107e84dca0e0bee1f8R1795-R1804).  It allows us to amortize the cost of an `mprotect` system call.

I can add all of ^^^ as a comment if it helps (the same info is in the commit message for this code)